### DISCUSSION TITLE
Positron-Server (Use positron in Web browser)?

### DISCUSSION THREAD

**Discussion by taiyodayo** [posit-dev/positron#4936] [Feedback] [answered]
In our team, we run RStudio server (rocker images) and Jupyterlab on our *big* machines (256gb ram, 32-core CPU, Cuda GPUs...),
so we can use low-specced MacBooks and still run occasional heavy tasks simply by opening Chrome.

I'd LOVE to add Positron in the mix!
is "positron-server", planned for release in future as OSS?

I hope use case like this does not conflict with the new policy.
(The usage is purely in-house. We have data that cannot go to public cloud.):

> you can't host Positron as a service to third parties 
[https://github.com/posit-dev/positron/discussions/3820](https://github.com/posit-dev/positron/discussions/3820)

**Comment by jthomasmock** [comment, marked as answer]
Hi @taiyodayo -- thanks for your interest!

Especially if most of your work can be done locally, you might be interested in our experimental support for remote SSH sessions: https://github.com/posit-dev/positron/pull/4251

Copying over some content from another response -- in short, no we are not planning to have a standalone "Positron server":

> we do not plan to release a stand-alone Positron server product. Instead, we plan to offer a desktop-to-server connection via Remote SSH (or equivalent functionality), i.e., a single-user connection where a user themselves can maintain the capabilities without being forced to only work on the server. In this workflow, there would be a headless installation of Positron components on a remote server, but Positron is otherwise fully functional from the desktop with some optional computing happening in a server for efficient scalability or data access purposes.

Alternatively, we are planning for preview support of Positron on Posit Workbench later this year, alongside RStudio, JupyterLab, and Code OSS (VS Code). Workbench has support for native sessions and jobs on traditional servers and HPC (Kubernetes, Slurm, Apptainer, Altair Grid Engine, etc).

**Comment by taiyodayo** [comment]
Hi @jthomasmock  thanks for the information!

In fact some of our R coders started using VSCode exactly for that ssh setup.
I can see with Positron everything can be much more streamlined!

While I see that it is already a great productivity boost, for "Ease of use" RStudio-on-web has been a great tool to onboard those new to R in our team.

Hope that decision not to release "Positron server" as OSS can change in future when other mainstream issues are sorted out.
it has been a great aid to introduce new hires quickly to the joys of data science and R.

Maybe a new paid-license model to allow us to run on our own machines?
(We have more budget than when we started out 5 years ago)

**Reply by ZacWarham** [reply]
A rocker style image would also be useful for us as we use those images in PrairieLearn to provide IDEs for students during their exams

**Comment by nfaux** [comment]
hi @jthomasmock, ssh connection is great, however having a server instance would be ideal for those cases where the requirement is to have a locked down server, where the data is sensitive, and thus cannot be moved. While one could set up a remote desktop instance, access via the web in a secure way, is best. This, way the network overhead is smaller and there is no requirement to move the data via ssh.
